CMakeLists.txt

```
@@ -104,7 +104,12 @@ option(TBBMALLOC_BUILD "Enable tbbmalloc build" ON)
 cmake_dependent_option(TBBMALLOC_PROXY_BUILD "Enable tbbmalloc_proxy build" ON "TBBMALLOC_BUILD" OFF)
 option(TBB_CPF "Enable preview features of the library" OFF)
 option(TBB_FIND_PACKAGE "Enable search for external oneTBB using find_package instead of build from sources" OFF)
-option(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH "Disable HWLOC automatic search by pkg-config tool" OFF)
+if(CMAKE_CROSSCOMPILING)
+    set(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH_DEFAULT ON)
+else()
+    set(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH_DEFAULT OFF)
+endif()
+option(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH "Disable HWLOC automatic search by pkg-config tool" ${TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH_DEFAULT})
```

Don't quite like the fact that we have `if/else` in the middle of the list of options: brings some inconvenience in reading this list. Maybe move this if/else above it or, since we rely only on `CMAKE_CROSSCOMPILING` to set this option's default value, make something like this:
```suggestion
option(TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH "Disable HWLOC automatic search by pkg-config tool" ${CMAKE_CROSSCOMPILING})
```